### ISSUE TITLE
1.8 aarch64 not running on m1/arm mac due to unsigned binary

### ISSUE THREAD

**Issue by daviehh** [JuliaLang/juliaup#278]
The aarch64 version of jula 1.8.0 beta is not able to launch on macos, while the x64 version runs ok. Running julia in terminal just errors with return code 1 with no output; running directly (instead of the launcher of juliaup, but run the julia binary downloaded by juliaup) errors with

```
 ~/.julia/juliaup/julia-1.8.0-beta1+0\~aarch64/bin/julia
[1]    57034 killed     ~/.julia/juliaup/julia-1.8.0-beta1+0\~aarch64/bin/julia
```

On a separate note, the arm version of 1.8.0-beta1 from `https://julialang.org/downloads/` runs normally. So, there seems to be some difference between the dmg in the download page and the binary used here

And from the github source's 1.8 branch:
- with `make` using all default parameters, the compiled binary runs normally
- however, if followed up with `make install`, the binary in the installed folder also fails with the same error message:

```
╰─❯ ./julia-1.8.0-beta1/bin/julia
[1]    10256 killed     ./julia-1.8.0-beta1/bin/julia
```

So looks like some copy/change link path errors during the `make install` stage?

Comparing the `~/.julia/juliaup/julia-1.8.0-beta1+0\~aarch64` folder with the jula.app in the dmg, not seeing missing files but the binary sizes of julia and the dylib files seems to differ a bit. Not sure if this issue is more suitable here or in the main julia repo?

**Comment by davidanthoff** [comment]
@staticfloat any idea?

**Comment by daviehh** [comment]
Looks like a codesign/security problem (copying signed file issue) for arm mac, running ok after ad-hoc sign for

```
codesign --sign - --force --preserve-metadata=entitlements,requirements,flags,runtime lib/libjulia.1.8.dylib
codesign --sign - --force --preserve-metadata=entitlements,requirements,flags,runtime bin/julia
```
maybe needed for all binaries/dylibs used?

ref: 
https://developer.apple.com/forums/thread/669145
https://stackoverflow.com/questions/67378106/mac-m1-cping-binary-over-another-results-in-crash

**Comment by daviehh** [comment]
Closing : fixed with 1.8.0-beta3 (2022-03-29)
